tests/test_manager_edgestore.py

```diff
@@ -0,0 +1,581 @@
+
+        self.assertIn(pmod_simple_hash, self.manager.object_cache_modification)
+
+        # Get
+        reloaded_pmod_simple_ls = self.manager.get_or_create_modification(self.graph, node_data)
+        self.assertEqual(pmod_simple_ls, reloaded_pmod_simple_ls)
+
+        # Create
+        node_data[VARIANTS] = [pmod_full]
+        pmod_full_ls = self.manager.get_or_create_modification(self.graph, node_data)
+        self.assertEqual(pmod_full, pmod_full_ls[0].data['mod_data'])
+
+        self.assertIn(pmod_full_hash, self.manager.object_cache_modification)
+
+        # Get
+        reloaded_pmod_full_ls = self.manager.get_or_create_modification(self.graph, node_data)
+        self.assertEqual(pmod_full_ls, reloaded_pmod_full_ls)
+
+        # Every modification was added only once to the object cache
+        self.assertEqual(8, len(self.manager.object_cache_modification.keys()))
+
+
+class TestDeletionCascade(BelReconstitutionMixin, TemporaryCacheMixin):
+    def setUp(self):
+        super(TestDeletionCascade, self).setUp()
+
+        @mock_bel_resources
+        def setup_graphs_and_networks(mock_get):
+            graph_a = pybel.from_path(path=test_bel_simple, manager=self.manager, allow_nested=True)
+            graph_b = pybel.from_path(path=test_bel_simple_b, manager=self.manager, allow_nested=True)
+            network_a = self.manager.insert_graph(graph_a)
+            network_b = self.manager.insert_graph(graph_b)
+            return graph_a, graph_b, network_a, network_b
+
+        self.graph_a, self.graph_b, self.network_a, self.network_b = setup_graphs_and_networks()
+        self.network_list = self.manager.list_networks()
+
+    def test_cascade_deletion_setup(self):
+        # 1. Check how many networks are present
+        self.assertEqual(2, self.manager.count_networks())
+        self.assertIn(self.network_a, self.network_list)
+        self.assertIn(self.network_b, self.network_list)
+
+        # 2. Check number of edges
+        self.assertEqual(8, self.manager.count_edges())
+
+        # 3. Check number of nodes
+        self.assertEqual(6, self.manager.count_nodes())
+
+        # 4. Check number of edge_properties
+
+        # 5. Check the node / edge overlap between networks
+        edges_in_a = self.network_a.edges.all()
+        edges_in_b = self.network_b.edges.all()
+
+        self.assertEqual(4, len(set(edges_in_a) - set(edges_in_b)))
+        self.assertEqual(2, len(set(edges_in_b) - set(edges_in_a)))
+        self.assertEqual(2, len(set(edges_in_a) & set(edges_in_b)))
+
+        nodes_in_a = self.network_a.nodes.all()
+        nodes_in_b = self.network_b.nodes.all()
+
+        self.assertEqual(1, len(set(nodes_in_a) - set(nodes_in_b)))
+        self.assertEqual(2, len(set(nodes_in_b) - set(nodes_in_a)))
+        self.assertEqual(3, len(set(nodes_in_a) & set(nodes_in_b)))
+
+        # 6. Check number of properties
+        self.assertEqual(1, self.manager.session.query(models.Property).count())
```

Comment:
since these count functions get used over and over, it's probably worth making them actual functions inside the manager

Reply:
Maybe an overall basic stats function that shows the numbers to all them tables? ^^

Reply:
yeah the other tables also have these basic count functions already